File: backend/src/test/java/ca/metricalsky/winston/test/annotations/ConverterTest.java
package ca.metricalsky.winston.test.annotations;

import ca.metricalsky.winston.convert.ConversionServiceAdapter;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mapstruct.extensions.spring.test.ConverterScan;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@ExtendWith(SpringExtension.class)
@ConverterScan(basePackageClasses = ConversionServiceAdapter.class)
public @interface ConverterTest {

}

File: backend/src/test/kotlin/ca/metricalsky/winston/test/annotation/ConverterTest.kt
package ca.metricalsky.winston.test.annotation

import io.kotest.core.extensions.ApplyExtension
import io.kotest.extensions.spring.SpringExtension
import org.mapstruct.extensions.spring.test.ConverterScan

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@ApplyExtension(SpringExtension::class)
@ConverterScan(basePackages = ["ca.metricalsky.winston.convert"])
annotation class ConverterTest
